apiserver/application/application.go

```diff
@@ -339,6 +340,73 @@ func (api *API) Update(args params.ApplicationUpdate) error {
 	return nil
 }
 
+// UpdateApplicationSeries updates the application series.  Series for
+// subordinates updated too.
+func (api *API) UpdateApplicationSeries(args params.ApplicationUpdateSeries) error {
+	if err := api.checkCanWrite(); err != nil {
+		return err
+	}
+	app, err := api.backend.Application(args.ApplicationName)
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	if !app.IsPrincipal() {
+		return errors.New(fmt.Sprintf("%s is a subordinate application, series update is not supported.", args.ApplicationName))
+	}
+
+	units, err := app.AllUnits()
+	if err != nil {
+		return errors.Trace(err)
+	}
+	// all units will have the same subordinates, therefore we only
+	// need to check one.
+	unit := units[0]
+	if unit != nil {
```

Comment:
When would unit ever be nil? I don't think we need this check. For safety, we should error though if  len(units) == 0